### ISSUE TITLE
Multivariate Legendre polynomials

### ISSUE THREAD

**Issue by iago-lito** [numpy/numpy#13120]
As found on the help pages for [Legendre](https://docs.scipy.org/doc/numpy-1.16.0/reference/routines.polynomials.legendre.html) module:

function | description
-- | --
legval(x, c[, tensor]) | Evaluate a Legendre series at points x.
legval2d(x, y, c) | Evaluate a 2-D Legendre series at points (x, y).
legval3d(x, y, z, c) | Evaluate a 3-D Legendre series at points (x, y, z).
leggrid2d(x, y, c) | Evaluate a 2-D Legendre series on the Cartesian product of x and y.
leggrid3d(x, y, z, c) | Evaluate a 3-D Legendre series on the Cartesian product of x, y, and z.
legroots(c) | Compute the roots of a Legendre series.
legfromroots(roots) | Generate a Legendre series with given roots.

Why is there not a `legvalNd(x, c, n)` to evaluate a n-D Legendre series at points x ∈ ℝ^n ?

Is there an efficient way to write it myself?

**Comment by eric-wieser** [comment]
Yes, such a function is very easy to write - take a look at #13107 for how those functions are implemented.

I've deliberately avoiding exposing `valnd` publicly yet, since in my opinion the polynomial API needs refining into an approach with a less repetitive API surface.
